project-plan.md

```diff
@@ -24,50 +24,95 @@ for programming because they facilitate exploratory programming and debugging.
 the client.
 
-## Methodologies and tooling
+# Methodologies and tooling
 
 During this project we will work using the SCRUM methodology with weekly
 sprints. Our backlog, current and completed sprints will be managed using
-Trello. Trello will also be used to communicate documents to the client
-and the TU coach.
-
-We will use pull-based development using GitHub. GitHub Issues will be used to
-track issues (all of this will be linked to Trello in one way or another). To
-facilitate this, we will be given our own repository within the Metaborg
-organization. Each of us will then fork this repository to our private GitHub
-accounts. We will also use one private repository to hold all the documents
+[Trello](https://trello.com/b/u2aKQ12y/bachelor-project-spoofax-repl).
+Trello will also be used to communicate documents to the client and the TU coach.
+Sprints will end on friday, coinciding with most deadlines according to the
+planning for this project as mandated by TU Delft.
+
+Each sprint will start with a "sprint planning event", in which user stories
+from the backlog will be assigned to project members, based on their
+prioritization and their feasibility for this sprint.
+Each sprint will end with a meeting in which we discuss to what degree
+we have achieved our sprint planning and how we can improve on that.
+A working demo should always be available at the time of the "sprint review".
+During sprints, the team will hold a daily scrum every day, where
+they will discuss their progress of the previous day and their planning for
+today, as well as any issues that could be detrimental to the weekly sprint goal.
+
+We will use a pull-based development model using GitHub in conjunction with
+the SCRUM methodology. GitHub Issues will be used to
+track issues (all of this will be linked to Trello in one way or another).
+To facilitate this, we will use repositories from our GitHub organization
+[spoofax-shell](https://github.com/spoofax-shell).
+Each of us will then fork these repositories to our private GitHub
+accounts. We will also use one repository to hold all the documents
 (e.g. this document, the research project, et cetera) supporting our project.
 
-Since Spoofax is an already existing project, we will use whichever tools are
+A pull request will be designated a status tag WIP, RFC or RDY:
+
+* WIP: Work in progress
+* RFC: Request for comments
+* RDY: Ready
+
+A pull request can be changed from WIP to RFC as soon as it is "done",
+which means it contains sufficient documentation and test coverage,
+and that the pull request passes continuous integration testing.
+Pull requests with status RFC should be reviewed by at least one other member
+of the development team. Once reviewed, the pull request can be changed
+to RDY, after which it can be merged.
+This process ensures that all code has been reviewed and tested before
+being merged into the final product, thereby always ensuring a working product.
+
```

Comment:
I think we should add a section here containing the following points (kindly taken from [Neovim](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md)):

## Submitting
* Make it clear in the issue tracker what you are working on.
* Be descriptive in your pull request description: what is it for, why is it needed, etc **or** link to the user story.
* Do not make cosmetic changes to unrelated files or sections in the same pull request. This creates noise, making reviews harder to do.

## Branching & history

* Do not work on your PR on the master branch, use a feature branch instead.
* Rebase your feature branch onto (upstream) master before opening the PR.
* Keep up to date with changes in (upstream) master so your PR is easy to merge.
* Try to actively tidy your history: combine related commits with interactive rebasing, separate monolithic commits, etc. If your PR is still [WIP], feel free to force-push to your feature branch to tidy your history.

Perhaps we should also have a section about [commit guidelines](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#commit-guidelines).